CHTCollectionViewWaterfallLayout.m

```diff
@@ -423,7 +423,9 @@ - (UICollectionViewLayoutAttributes *)layoutAttributesForSupplementaryViewOfKind
 - (NSArray *)layoutAttributesForElementsInRect:(CGRect)rect {
   NSInteger i;
   NSInteger begin = 0, end = self.unionRects.count;
-  NSMutableArray *attrs = [NSMutableArray array];
+  NSMutableDictionary *cellAttrDict = [NSMutableDictionary dictionary];
+  NSMutableDictionary *supplAttrDict = [NSMutableDictionary dictionary];
+  NSMutableDictionary *decorAttrDict = [NSMutableDictionary dictionary];
```

Comment:
Do we really need 3 dictionaries? Or what we need is one dictionary and use `attr.indexPath` as the key?